internal/services/loganalytics/log_analytics_workspace_resource.go

```diff
@@ -154,6 +154,11 @@ func resourceLogAnalyticsWorkspace() *pluginsdk.Resource {
 				ValidateFunc: datacollectionrules.ValidateDataCollectionRuleID,
 			},
 
+			"immediate_purge_data_on_30_days_enabled": {
```

Comment:
Can we rename this to be grammatically correct
```suggestion
			"immediate_data_purge_on_30_days_enabled": {
```